process/transaction/shardProcess.go

```diff
@@ -171,6 +174,40 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) (vmco
 	return vmcommon.UserError, process.ErrWrongTransaction
 }
 
+func (txProc *txProcessor) executeAfterFailedMoveBalanceTransaction(
+	tx *transaction.Transaction,
+	txError error,
+) error {
+	acntSnd, err := txProc.getAccountFromAddress(tx.SndAddr)
+	if err != nil {
+		return err
+	}
+
+	switch txError {
+	case process.ErrInvalidMetaTransaction:
+		return txProc.executingFailedTransaction(tx, acntSnd, txError)
+	case process.ErrAccountNotPayable:
+		snapshot := txProc.accounts.JournalLen()
+		txHash, err := core.CalculateHash(txProc.marshalizer, txProc.hasher, tx)
+		if err != nil {
+			return err
+		}
+
+		err = txProc.scProcessor.ProcessIfError(acntSnd, txHash, tx, txError.Error(), nil, snapshot)
+		if err != nil {
+			return err
+		}
+
+		if check.IfNil(acntSnd) {
```

Comment:
Is this possible at this point? Perhaps moved earlier for safety?

Reply:
yes - in case of cross shard transactions destination this shard.